### PR TITLE
[observability][autoscaler] Ensure pending nodes is reset to 0 after scaling

### LINKED ISSUE #31982: [Dashboard] dashboard showing stale pending nodes while the cluster is fully autoscaled.
### What happened + What you expected to happen

start the [cluster](https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_u9jnexvbz53n9vzfjgdpwu9d71?command-history-section=head_start_up_log) with 64 nodes. you will notice even it autoscaling finished, the number of pending nodes doesn't decrease to 0.

image

### Versions / Dependencies

master

### Reproduction script

run release test `many_tasks`. 
or run `python test_many_tasks.py --num-tasks=10000` on https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_u9jnexvbz53n9vzfjgdpwu9d71?command-history-section=head_start_up_log

### Issue Severity

None

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

The previous way pending_nodes was calculated was prone to race conditions, instead, let's just always publish it in the main thread with other metrics.

## Related issue number

Closes #31982

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
cc @gvspraveen

**Review by rkooo567 — APPROVED** [review]

File: python/ray/tests/test_autoscaler_e2e.py [hunk 1]
```diff
@@ -91,7 +91,11 @@ def ping(self):
                 return True
 
         zero_reported_condition = get_metric_check_condition(
-            {"autoscaler_cluster_resources": 0, "autoscaler_pending_resources": 0},
+            {
+                "autoscaler_cluster_resources": 0,
+                "autoscaler_pending_resources": 0,
+                "autoscaler_pending_nodes": 0,
```
Comment (rkooo567) [inline python/ray/tests/test_autoscaler_e2e.py:97]: Any way to write a test that could've caught this?

**Review by rkooo567 — APPROVED** [review]

File: python/ray/tests/test_autoscaler_e2e.py (hunk 1 above)
Comment (wuisawesome) [reply python/ray/tests/test_autoscaler_e2e.py:97]: isn't this precisely a test that catches this?

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by wuisawesome** [comment]
Remaining test failures look unrelated/flakey. merging

**Pull request merged**